packages/backend-core/src/features/index.ts

```diff
@@ -159,11 +159,11 @@ export class FlagSet<V extends Flag<any>, T extends { [key: string]: V }> {
             specificallySetFalse.add(feature)
           }
 
-          if (!this.isFlagName(feature)) {
-            throw new Error(`Feature: ${feature} is not an allowed option`)
```

Comment:
I'd just change this one line to be `continue` instead of combining with the next conditional, think it makes the intent a little bit clearer.

Also the way you've done it here I think will push junk flags into flagValues and it's not been caught because that line has a `@ts-expect-error` above it.